### PR TITLE
fix: handle more scenarios for stream reconnection

### LINKED ISSUE #427: Closing the stream because it has been inactive for 600 seconds
Thanks for stopping by to ask us a question! Please make sure to include:
- What you're trying to do

> I'm trying to write to BQ Table via storage API . I have used WriterClient to createSteamConnection and I append rows once the stream is created. 

- What code you've already tried
```
const connection = await this.client.write.createStreamConnection({
                streamType,
                destinationTable
            });
            let instance = this;
            connection.onConnectionError((err)=>{
                console.log('err has happened',err)
                instance.init(req);
            })

            const streamId = connection.getStreamId();

            const writer = new Writer({
                streamId,
                connection,
                protoDescriptor

            });
then later on 

writer.appendRows({ serializedRows: rows });

```
`
- Any error messages you're getting
  **code: 10**,
  details: 'Closing the stream because it has been inactive for 600 seconds.

**PLEASE READ**: If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/) instead of filing on GitHub. This will ensure a timely response.

### PR DESCRIPTION
**Pull request by alvarowolfx**
The BigQuery service now return Code:10 Aborted when a connection is idle for to long, to force reconnection.

Fixes #427  🦕

### CODE REVIEW

File: src/managedwriter/stream_connection.ts [hunk 1]
```diff
@@ -135,15 +135,18 @@ export class StreamConnection extends EventEmitter {
   private shouldReconnect(err: gax.GoogleError): boolean {
     if (
       err.code &&
-      [gax.Status.UNAVAILABLE, gax.Status.RESOURCE_EXHAUSTED].includes(
-        err.code
-      ) &&
+      [
+        gax.Status.UNAVAILABLE,
```
Comment (shollyman) [inline src/managedwriter/stream_connection.ts:139]: Let's also include CANCELLED and DEADLINE_EXCEEDED here.

**Review by shollyman — APPROVED** [review]
Minor request to widen the predicate, otherwise looks good.

File: system-test/managed_writer_client_test.ts [hunk 2]
```diff
@@ -780,6 +784,91 @@ describe('managedwriter.WriterClient', () => {
...
+            code: gax.Status.ABORTED,
+            msg: 'Closing the stream because it has been inactive',
+          },
+          {
+            code: gax.Status.RESOURCE_EXHAUSTED,
+            msg: 'read econnreset',
+          },
+          {
+            code: gax.Status.ABORTED,
+            msg: 'service is currently unavailable', // schema mismatch
```
Comment (shollyman) [inline system-test/managed_writer_client_test.ts:835]: nit: stale comment about mismatch?

**Review by shollyman — APPROVED** [review]

File: system-test/managed_writer_client_test.ts (hunk 2 above)
Comment (alvarowolfx) [reply system-test/managed_writer_client_test.ts]: removed

**Review by feywind — APPROVED** [review]

**Pull request merged**